test/unit/transformRemappings.spec.ts

```diff
@@ -5,35 +5,137 @@ import mock from 'mock-fs';
+      import 'node_modules/another-package/Token.sol';
+      import {FixedPointMathLib} from 'node_modules/solmate/src/utils/FixedPointMathLib.sol';
+      import {FixedPointMathLib} from "node_modules/solmate/src/utils/FixedPointMathLib.sol";
+      import {FixedPointMathLib as FPL} from "node_modules/solmate/src/utils/FixedPointMathLib.sol";
+      import * as FPL from "../node_modules/solmate/src/utils/FixedPointMathLib.sol";
+    `;
+
+      const transformedContent = transformRemappings(fileContent);
+
+      expect(transformedContent).to.include(`import 'some-package/Contract.sol';`);
+      expect(transformedContent).to.include(`import 'another-package/Token.sol';`);
+      expect(transformedContent).to.include(`import {FixedPointMathLib} from 'solmate/src/utils/FixedPointMathLib.sol';`);
+      expect(transformedContent).to.include(`import {FixedPointMathLib} from "solmate/src/utils/FixedPointMathLib.sol";`);
+      expect(transformedContent).to.include(`import {FixedPointMathLib as FPL} from "solmate/src/utils/FixedPointMathLib.sol";`);
+      expect(transformedContent).to.include(`import * as FPL from "solmate/src/utils/FixedPointMathLib.sol";`);
+    });
+  });
+
+  describe('with remapping for path inside node_modules', function () {
+    before(() => {
+      mock({
+        'remappings.txt': `
+          solmate/=node_modules/solmate/src/`,
+      });
+    });
+
+    it('should apply remapping and then remove node_modules from path', () => {
+      const fileContent = `
+                                  import {FixedPointMathLib} from 'solmate/utils/FixedPointMathLib.sol';
+                                  import 'solmate/utils/FixedPointMathLib.sol';
+                                  import "solmate/utils/FixedPointMathLib.sol";
+                                  import {FixedPointMathLib as FPL} from 'solmate/utils/FixedPointMathLib.sol';
+                                  import * as FPL from "solmate/utils/FixedPointMathLib.sol";
+                                `;
+
+      const transformedContent = transformRemappings(fileContent);
+
+      expect(transformedContent).to.include(`import {FixedPointMathLib} from 'solmate/src/utils/FixedPointMathLib.sol';`);
+      expect(transformedContent).to.include(`import 'solmate/src/utils/FixedPointMathLib.sol';`);
+      expect(transformedContent).to.include(`import "solmate/src/utils/FixedPointMathLib.sol";`);
+      expect(transformedContent).to.include(`import {FixedPointMathLib as FPL} from 'solmate/src/utils/FixedPointMathLib.sol';`);
+      expect(transformedContent).to.include(`import * as FPL from "solmate/src/utils/FixedPointMathLib.sol";`);
+    });
+  });
 
-  it('should transform imports with remappings', () => {
+  it('should leave unrelated imports alone', () => {
     const fileContent = `
-        import '../../../interfaces/ITest.sol';
-        import './someFile.sol';
-      `;
+      import './Contract.sol';
```

Comment:
what do you think about adding something that should be automatically modified in between these lines? Just to make sure our code is not skipping all the imports after seeing that the first one does not need any modification

No need to add a new test, we could do it in this one

Reply:
sure!

now that I think about it (related to your other comment) having multiple imports in a single line will probably break as well  (but we could actually make that work without properly parsing the code). Will add a test for that as well

Reply:
a226bca